ludwig/api.py

```diff
@@ -213,10 +213,13 @@ def __init__(
             config_dict = copy.deepcopy(config)
             self.config_fp = None
 
+        self.base_config = config_dict
+
         # Upgrades deprecated fields and adds new required fields in case the config loaded from disk is old.
-        self.base_config = upgrade_to_latest_version(config_dict)
+        upgraded_config = upgrade_to_latest_version(copy.deepcopy(config_dict))
```

Comment:
upgrade_to_latest_version returns a copy, doesn't modify in-place.
```suggestion
        upgraded_config = upgrade_to_latest_version(config_dict)
```